src/api/Base.js

```diff
@@ -123,6 +134,78 @@ class Base {
     getCache(): Cache {
         return this.cache;
     }
+
+    /**
+     * Generic success handler
+     *
+     * @param {Object} data the response data
+     * @param {Function} successCallback the success callback
+     */
+    successHandler(data: Object): void {
+        if (!this.isDestroyed() && this.successCallback && typeof this.successCallback === 'function') {
+            this.successCallback(data);
+        }
+    }
+
+    /**
+     * Generic error handler
+     *
+     * @param {Object} data the response data
+     * @param {Function} errorCallback the error callback
+     */
+    errorHandler(error: Object): void {
+        if (!this.isDestroyed() && this.errorCallback && typeof this.errorCallback === 'function') {
+            this.errorCallback(error);
```

Comment:
The signature of this is different than what it was in Item.js
error in the args, I think it will be a Error event.
This event will have the response, which in turn will have the data.
I think you want to send that data in the errorCallback

Reply:
I think I removed it because I didnt see any error handling other than a console.log. Also, returning the raw error object could potentially be more helpful for debugging, since I assume console messages will be stripped in production bundle

Reply:
Changed this to be like the logic in Item.js